psga/s_aureus/psga.nf

```diff
@@ -20,18 +21,25 @@ workflow psga {
         // collect waits for all samples to finish. generate_s_aureus_output only runs once
         generate_s_aureus_output(
             ch_metadata,
-            bactopia_one.out.ch_input_files.collect(),
-            bactopia_one.out.annotation_summary.collect(),
-            bactopia_one.out.variants_txt_for_csv_file.collect(),
-            bactopia_one.out.all_software_versions.collect(),
-            bactopia_one.out.checkm_results_txt.collect(),
-            bactopia_one.out.assembly_json.collect(),
-            bactopia_one.out.antimicrobial_protein_report.collect(),
-            bactopia_one.out.antimicrobial_gene_report.collect(),
-            bactopia_one.out.mykrobe_versions.collect(),
-            bactopia_one.out.mlst_versions.collect()
+            bactopia_one.out.ch_input_files,
+            bactopia_one.out.annotation_summary,
+            bactopia_one.out.variants_txt_for_csv_file,
+            bactopia_one.out.all_software_versions,
+            bactopia_one.out.checkm_results_txt,
+            bactopia_one.out.assembly_json,
+            bactopia_one.out.antimicrobial_protein_report,
+            bactopia_one.out.antimicrobial_gene_report,
+            bactopia_one.out.mykrobe_versions,
+            bactopia_one.out.mlst_versions,
+            bactopia_one.out.genome_assembly_name
```

Comment:
You are the assumption that these nextflow channels are somehow linked by sample (e.g. the channels `bactopia_one.out.annotation_summary` and `bactopia_one.out.ch_input_files` are organised so that when you call them you have content consistent with the same sample.

Have you checked that this is the case when running multi samples?

Channels are in principle `detached` from each other in order to enable parallelism.

The previous code does not have this issue because the results for all samples will be available to the process.
The new code can have results from different samples mixed together.

Reply:
That's a good catch. I thought they were linked by sample.

I need to fix it now

Reply:
Do you have a link for how to combine channels, or is there somewhere we do it in code?

> one single channel of type tuple

Reply:
Fixed